packages/workspace/src/workspace/workspace.ts

```diff
@@ -684,25 +684,33 @@ export const loadWorkspace = async (
 
     const updateWorkspace = async (envName: string): Promise<void> => {
       const source = naclFilesSource
-      const getElementsDependents = async (elemIDs: ElemID[], addedIDs: Set<string>): Promise<ElemID[]> => {
-        elemIDs.forEach(id => addedIDs.add(id.getFullName()))
-        const filesWithDependencies = _.uniq(
-          await awu(elemIDs)
-            .flatMap(id => source.getElementReferencedFiles(envName, id))
-            .toArray(),
-        )
-        const dependentsIDs = _.uniqBy(
-          await awu(filesWithDependencies)
-            .map(filename => source.getParsedNaclFile(filename))
-            .flatMap(async naclFile => ((await naclFile?.elements()) ?? []).map(elem => elem.elemID))
-            .filter(id => !addedIDs.has(id.getFullName()))
-            .toArray(),
-          id => id.getFullName(),
+      const getElementsDependents = async (elemIDs: ElemID[], addedIDs: Set<string>): Promise<ElemID[]> =>
+        log.timeDebug(
+          async () => {
+            elemIDs.forEach(id => addedIDs.add(id.getFullName()))
+            const filesWithDependencies = _.uniq(
+              await awu(elemIDs)
+                .flatMap(id => source.getElementReferencedFiles(envName, id))
+                .toArray(),
+            )
```

Comment:
worth adding time log here too
```suggestion
            const filesWithDependencies =  await log.timeDebug(
              async () => _.uniq(
                await awu(elemIDs)
                  .flatMap(id => source.getElementReferencedFiles(envName, id))
                  .toArray(),
              ),
              'running getElementReferencedFiles for %s elemIds',
              elemIDs.length,
            )
```